src/Sparrow.Server/Utils/VxSort/codegen/bitonic_avx2.py

```diff
@@ -0,0 +1,652 @@
+import os
+from datetime import datetime
+
+from utils import native_size_map, next_power_of_2, autogenerated_blabber
+from bitonic_isa import BitonicISA
+
+
+class AVX2BitonicISA(BitonicISA):
+    def __init__(self, type, configuration):
+        self.vector_size_in_bytes = 32
+
+        self.type = type
+
+        self.bitonic_size_map = {}
+
+        for t, s in native_size_map.items():
+            self.bitonic_size_map[t] = int(self.vector_size_in_bytes / s)
+
+        self.bitonic_type_map = {
+            "int": "Int32",
+            "uint": "Int32",
+            "float": "Int32",
+            "long": "Int64",
+            "ulong": "Int64",
+            "double": "Int64",
+        }
+
+        self._max_bitonic_sort_vectors = configuration.max_bitonic_sort_vectors
+        self.unroll_bitonic_sorters = configuration.unroll_bitonic_sorters
+        self.namespace = configuration.namespace
+
+        self._var_names = [f"d{i:02d}" for i in range(0, self.max_bitonic_sort_vectors + 2)]
+
+    @property
+    def max_bitonic_sort_vectors(self):
+        return self._max_bitonic_sort_vectors
+
+    def vector_size(self):
+        return self.bitonic_size_map[self.type]
+
+    def vector_type(self):
+        return "V"
+
+    @classmethod
+    def supported_types(cls):
+        return native_size_map.keys()
+
+    def i2d(self, v):
+        t = self.type
+        if t == "double":
+            return v
+        return f"Vector256.AsDouble({v})"
+
+    def i2s(self, v):
+        t = self.type
+        if t == "double":
+            raise Exception("Incorrect Type")
+        elif t == "float":
+            return f"Vector256.AsSingle({v})"
+        return v
+
+    def d2i(self, v):
+        t = self.type
+        if t == "float":
+            return f"Vector256.AsSingle({v})"
+        elif t == "int":
+            return f"Vector256.AsInt32({v})"
+        elif t == "uint":
+            return f"Vector256.AsUInt32({v})"
+        elif t == "long":
+            return f"Vector256.AsInt64({v})"
+        elif t == "ulong":
+            return f"Vector256.AsUInt64({v})"
+
+        return v
```

Comment:
@damageboy This is the concept of composability, had to move this stuff back into the generator.